### ISSUE TITLE
Automatic scrolling on arrow doesn't work when using the default slot

### ISSUE THREAD

**Issue by jimmyhogoboom** [trevoreyre/autocomplete#78]
Normally when using the arrow keys to scroll through the results, if you reach the end of the container it will automatically scroll to keep the next selection in view, as seen [here](https://autocomplete.trevoreyre.com/#/vue-component?id=result):
![working](https://user-images.githubusercontent.com/61669826/78054643-fd6ef100-7347-11ea-90cb-08d74f0e7b33.gif)

However when using the default slot to define our own markup, the results container is not automatically scrolled, as seen [here](https://autocomplete.trevoreyre.com/#/vue-component?id=default-slot):
![not working](https://user-images.githubusercontent.com/61669826/78054927-6b1b1d00-7348-11ea-9f34-2aae5c8e402e.gif)

These examples were taken directly from [the documentation page](https://autocomplete.trevoreyre.com/#/vue-component?id=default-slot) in Chrome 80.

**Comment by trevoreyre** [comment]
Yep good call. Keeping the selected element in view relies on a ref to the list element, which doesn't get preserved if you're using the default slot. I'll have to look into how to refactor that to work a different way.

**Comment by jardakotesovec** [comment]
@trevoreyre I believe for this use case composition api will come to rescue :-). Have you considered to ship autocomplete component with composition api? I think it would be great fit for people they need these custom layouts/visuals instead of slots.  

Alternative for refs here would be unfortunately just querying for the elements instead of using refs directly.

**Comment by trevoreyre** [comment]
@jardakotesovec I haven't had a chance to dig too deep into the composition API yet, other than a few super basic examples. But I'm super interested to learn more! Do you happen to have any specific documentation or articles to point me in the right direction about how composition API might help solve this issue?

**Comment by xscasnya** [comment]
Do you plan to include fix for this in the next release ?

**Comment by ednelson89** [comment]
Not sure if there's been any work on this or not; is there a work-around for this that allows for customizing the input without accessing the #default slot?

**Comment by ednelson89** [comment]
So, I finally got around to resolving this on my project: I realized that the input has the aria-activedescendant attribute that changes to reflect the highlighted option, so I used a MutationObserver in the mounted lifecycle hook to watch that attribute, and that used some generic scrolling logic to target those list-items. Implementing a Mutation Observer can be found on MDN,
and the callback function that it uses to handle the scrolling @attribute-change is below. There may be some optimizations you can do, but it seems to work pretty well:

`manageScroll(mutationsList, observer) {
                       // You'll need to be tracking the input's ID somewhere in your file so you can use it to target the overall parent input
  		       let temporaryElement = document.getElementById(this.inputId)
  		       mutationsList.forEach((record) => {
  		         if (record.attributeName === 'aria-activedescendant') {
                          
                           // Get your focused list item
  		           let focusedListItem = temporaryElement.getAttribute('aria-activedescendant')
  		           let selectedOption = document.getElementById(focusedListItem)

                           // Get the list box itself
  		           let owns = temporaryElement .getAttribute('aria-owns')
  		           let container = document.getElementById(owns)
  		  
  		           if (focusedListItem && container.scrollHeight > container.clientHeight) {
  		             var scrollBottom = container.clientHeight + container.scrollTop
  		             var elementBottom = selectedOption.offsetTop + selectedOption.offsetHeight
  		             if (elementBottom > scrollBottom) {
  		               container.scrollTop = elementBottom - container.clientHeight
  		             } else if (selectedOption.offsetTop < container.scrollTop) {
  		               container.scrollTop = selectedOption.offsetTop
  		             }
  		           }
  		         }
  		       })`
